### ISSUE TITLE
`do catch` is accidentally stable in 2018 edition

### ISSUE THREAD

**Issue by scottmcm** [rust-lang/rust#52604]
This works with no `#![feature]`:
```rust
fn main() {
    let x: Option<_> = do catch { 4 };
}
```
https://play.rust-lang.org/?gist=ed21b1848a1114ed7f7fefc13bdb8880&version=nightly&mode=debug&edition=2018

I expect this to be fixed with https://github.com/rust-lang/rust/pull/52602, but wanted to file explicitly for tracking.
